diff --git a/cmake/ProcessOptions.cmake b/cmake/ProcessOptions.cmake
index fc4bd449d..238c04c6e 100644
--- a/cmake/ProcessOptions.cmake
+++ b/cmake/ProcessOptions.cmake
@@ -374,6 +374,7 @@ function( NEST_PROCESS_WITH_PYTHON )
     else()
       find_package( Python 3.8 REQUIRED Interpreter Development.Module )
     endif()
+    find_package( PythonLibs )
 
     if ( Python_FOUND )
       if ( CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT )
