diff --git a/models/spike_generator.h b/models/spike_generator.h
index 3939f24d5..e523bb12f 100644
--- a/models/spike_generator.h
+++ b/models/spike_generator.h
@@ -284,7 +284,7 @@ private:
     //! Shift spike times at present to next step
     bool shift_now_spikes_;
 
-    Parameters_();                               //!< Sets default parameter values
+    Parameters_(); //!< Sets default parameter values
     Parameters_( const Parameters_& ) = default;
     Parameters_& operator=( const Parameters_& ) = default;
 
diff --git a/nestkernel/CMakeLists.txt b/nestkernel/CMakeLists.txt
index 916fd92bf..4e5743546 100644
--- a/nestkernel/CMakeLists.txt
+++ b/nestkernel/CMakeLists.txt
@@ -130,6 +130,9 @@ if ( HAVE_MPI )
   )
 endif ()
 
+# Prevent problems with Conda path substitution (see #2348)
+set_source_files_properties( dynamicloader.cpp PROPERTIES COMPILE_OPTIONS "-O0" )
+
 
 add_library( nestkernel ${nestkernel_sources} )
 target_link_libraries( nestkernel
diff --git a/nestkernel/dynamicloader.cpp b/nestkernel/dynamicloader.cpp
index c239d1341..009f9e2d3 100644
--- a/nestkernel/dynamicloader.cpp
+++ b/nestkernel/dynamicloader.cpp
@@ -270,7 +270,11 @@ DynamicLoaderModule::init( SLIInterpreter* i )
     LOG( M_ERROR, "DynamicLoaderModule::init", "Could not initialize libltdl. No dynamic modules will be available." );
   }
 
-  if ( lt_dladdsearchdir( NEST_INSTALL_PREFIX "/" NEST_INSTALL_LIBDIR ) )
+  // To avoid problems due to string substitution in NEST binaries during
+  // Conda installation, we need to convert the literal to string, cstr and back,
+  // see #2237 and https://github.com/conda/conda-build/issues/1674#issuecomment-280378336
+  const std::string module_dir = std::string( NEST_INSTALL_PREFIX ).c_str() + std::string( "/" NEST_INSTALL_LIBDIR );
+  if ( lt_dladdsearchdir( module_dir.c_str() ) )
   {
     LOG( M_ERROR, "DynamicLoaderModule::init", "Could not add dynamic module search directory." );
   }
diff --git a/sli/CMakeLists.txt b/sli/CMakeLists.txt
index 18248ee55..a2b09c84a 100644
--- a/sli/CMakeLists.txt
+++ b/sli/CMakeLists.txt
@@ -82,6 +82,10 @@ set( sli_sources
     utils.cc utils.h
     )
 
+
+# Prevent problems with Conda path substitution (see #2348)
+set_source_files_properties( slistartup.cc PROPERTIES COMPILE_OPTIONS "-O0" )
+
 add_library( sli_lib ${sli_sources} )
 target_link_libraries( sli_lib nestutil OpenMP::OpenMP_CXX )
 
diff --git a/sli/slistartup.cc b/sli/slistartup.cc
index 271d4d0f6..dea5771fc 100644
--- a/sli/slistartup.cc
+++ b/sli/slistartup.cc
@@ -133,9 +133,11 @@ SLIStartup::GetenvFunction::execute( SLIInterpreter* i ) const
   i->EStack.pop();
 }
 
-
 SLIStartup::SLIStartup( int argc, char** argv )
-  : sliprefix( NEST_INSTALL_PREFIX )
+  // To avoid problems due to string substitution in NEST binaries during
+  // Conda installation, we need to convert the literal to string, cstr and back,
+  // see #2237 and https://github.com/conda/conda-build/issues/1674#issuecomment-280378336
+  : sliprefix( std::string( NEST_INSTALL_PREFIX ).c_str() )
   , slilibdir( sliprefix + "/" + NEST_INSTALL_DATADIR )
   , slidocdir( sliprefix + "/" + NEST_INSTALL_DOCDIR )
   , startupfile( slilibdir + "/sli/sli-init.sli" )
