diff --git a/cmake/ProcessOptions.cmake b/cmake/ProcessOptions.cmake
index e20c3086df..b93e84ea8b 100644
--- a/cmake/ProcessOptions.cmake
+++ b/cmake/ProcessOptions.cmake
@@ -368,6 +368,7 @@ function( NEST_PROCESS_WITH_PYTHON )
     else()
       find_package( Python 3.8 REQUIRED Interpreter Development.Module )
     endif()
+    find_package( PythonLibs )
 
     if ( Python_FOUND )
       if ( CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT )
