diff --git a/doc/htmldoc/conf.py b/doc/htmldoc/conf.py
index acd240883..7391562a0 100644
--- a/doc/htmldoc/conf.py
+++ b/doc/htmldoc/conf.py
@@ -43,6 +43,8 @@ else:
 
 if os.environ.get("READTHEDOCS") == "True":
     doc_build_dir = source_dir / "doc/htmldoc"
+elif os.environ.get("CONDA_BUILD") == 1:
+    doc_build_dir = source_dir / "doc/htmldoc"
 else:
     doc_build_dir = Path(os.environ["OLDPWD"]) / "doc/htmldoc"
 
